File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n+1);

        dp[0]=nums[0];
        dp[1]=max(nums[0],nums[1]);
        dp[2]=max(dp[1],dp[0]+nums[2]);

        for(int i=3;i<n;i++){
            dp[i]=max(dp[i-1],dp[i-2]+nums[i]);
        }

        return dp[n-1];

    }
};

File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n = len(intervals)
        l = 0 

        intervals.sort()
        res = []
        while l<n-1:
            s,e = intervals[l]
            s1,e1 = intervals[l+1]

            if e<s1:
                res.append([s,e])
                l+=1
                continue
            else:
                s = min(s,s1)
                e = max(e,e1)
                intervals[l+1] = [s,e]
                l+=1
        res.append(intervals[n-1])
        return res        

